inventory-api/src/main/scala/com/normation/inventory/services/provisionning/CheckInventoryDigest.scala

```diff
@@ -152,7 +152,8 @@ class InventoryDigestServiceV1(
   def getKey (receivedInventory  : InventoryReport) : Box[(PublicKey, KeyStatus)] = {
     for {
       nodeInventory <- repo.get(receivedInventory.node.main.id) match {
-        case Full(storedInventory) =>
+        // Don't get the node inventory if there is no public key in it
+        case Full(storedInventory) if ! storedInventory.node.publicKeys.isEmpty =>
```

Comment:
I'm wondering if we should not also try to read the key status in the stored inventory, so that one does not have to just corrupt the strored key to have the new one accepter, he also need to change the status back to not verified.